### PR TITLE
pkg/metaserver: optimize key.go

### PR DESCRIPTION
**Pull request by wzshiming**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/metaserver/key.go [hunk 1]
```diff
@@ -30,30 +30,26 @@ func KeyFuncObj(obj runtime.Object) (string, error) {
...
+
+	objKind := obj.GetObjectKind()
+	gvk := objKind.GroupVersionKind()
+	if gvk.Empty() {
 		return "", fmt.Errorf("could not get group/version/kind information in obj")
 	}
-	group := gvk.GroupVersionKind().Group
-	version := gvk.GroupVersionKind().Version
-	resources := util.UnsafeKindToResource(gvk.GroupVersionKind().Kind)
+	group := gvk.Group
```
Comment (daixiang0) [inline pkg/metaserver/key.go:39]: Could we reduce temp val?

**Comment by daixiang0** [comment]
Merge with #2724 please.

File: pkg/metaserver/key.go (hunk 1 above)
Comment (wzshiming) [reply pkg/metaserver/key.go:39]: `GroupVersionKind()` is not a zero-cost function in most implementations, I don't think it is necessary to use it multiple times.
https://github.com/kubeedge/kubeedge/blob/f30b2021cded5191aa116f8ff51dbc9dc1f677d3/vendor/k8s.io/apimachinery/pkg/runtime/schema/group_version.go#L296-L305

File: pkg/metaserver/key.go [hunk 2]
```diff
@@ -30,68 +30,60 @@ func KeyFuncObj(obj runtime.Object) (string, error) {
...
+	namespaces := info.Namespace
+	name := info.Name
+	if namespaces == "" {
+		namespaces = v2.NullNamespace
+	}
+	if name == "" {
+		name = v2.NullName
+	}
+
+	key := "/" + group + "/" + version + "/" + resource + "/" + namespaces + "/" + name
```
Comment (daixiang0) [inline pkg/metaserver/key.go:86]: Better to use string.Join or fmt.Sprintf

Comment (wzshiming) [reply pkg/metaserver/key.go]: Thanks, updated

**Review by GsssC — COMMENTED** [review]
/lgtm thx for contribution! : )

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**